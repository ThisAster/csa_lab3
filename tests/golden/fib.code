[{"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 1},
 {"opcode": "ld", "arg_type": "immediate", "arg": 2},
 {"opcode": "st", "arg_type": "address", "arg": 2},
 {"opcode": "ld", "arg_type": "immediate", "arg": 4000000},
 {"opcode": "st", "arg_type": "address", "arg": 3},
 {"opcode": "ld", "arg_type": "address", "arg": 2},
 {"opcode": "sub", "arg_type": "address", "arg": 3},
 {"opcode": "jle", "arg_type": "address", "arg": 13},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 14},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 34},
 {"opcode": "ld", "arg_type": "address", "arg": 2},
 {"opcode": "mod", "arg_type": "immediate", "arg": 2},
 {"opcode": "sub", "arg_type": "immediate", "arg": 0},
 {"opcode": "je", "arg_type": "address", "arg": 21},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 22},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 26},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 2},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 4},
 {"opcode": "ld", "arg_type": "address", "arg": 2},
 {"opcode": "st", "arg_type": "address", "arg": 1},
 {"opcode": "ld", "arg_type": "address", "arg": 4},
 {"opcode": "add", "arg_type": "address", "arg": 2},
 {"opcode": "st", "arg_type": "address", "arg": 2},
 {"opcode": "jmp", "arg_type": "address", "arg": 8},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 6},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "sub", "arg_type": "immediate", "arg": 0},
 {"opcode": "jne", "arg_type": "address", "arg": 41},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 42},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 54},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "mod", "arg_type": "immediate", "arg": 10},
 {"opcode": "st", "arg_type": "address", "arg": 5},
 {"opcode": "ld", "arg_type": "address", "arg": 6},
 {"opcode": "mul", "arg_type": "immediate", "arg": 10},
 {"opcode": "add", "arg_type": "address", "arg": 5},
 {"opcode": "st", "arg_type": "address", "arg": 6},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "div", "arg_type": "immediate", "arg": 10},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 36},
 {"opcode": "ld", "arg_type": "address", "arg": 6},
 {"opcode": "sub", "arg_type": "immediate", "arg": 0},
 {"opcode": "jne", "arg_type": "address", "arg": 59},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 60},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 71},
 {"opcode": "ld", "arg_type": "address", "arg": 6},
 {"opcode": "mod", "arg_type": "immediate", "arg": 10},
 {"opcode": "st", "arg_type": "address", "arg": 5},
 {"opcode": "ld", "arg_type": "immediate", "arg": 48},
 {"opcode": "add", "arg_type": "address", "arg": 5},
 {"opcode": "out", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 6},
 {"opcode": "div", "arg_type": "immediate", "arg": 10},
 {"opcode": "st", "arg_type": "address", "arg": 6},
 {"opcode": "jmp", "arg_type": "address", "arg": 54},
 {"opcode": "hlt", "arg_type": "none", "arg": 0}]