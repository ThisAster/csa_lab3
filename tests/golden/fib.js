var sumeven
sumeven = 0
var prev
prev = 1
var current
current = 2
var limit
limit = 4000000
var tempvar
while (current <= limit) {
    if (current % 2 == 0) {
        sumeven = sumeven + current
    }
    tempvar = prev
    prev = current
    current = tempvar + current
}

var digit
var sumreversed
sumreversed = 0
while (sumeven != 0) {
    digit = sumeven % 10
    sumreversed = sumreversed * 10 + digit
    sumeven = sumeven / 10
}

while (sumreversed != 0) {
    digit = sumreversed % 10
    print(48 + digit)
    sumreversed = sumreversed / 10
}
