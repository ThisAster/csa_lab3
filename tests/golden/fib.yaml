js: fib.js
code: fib.code
data: fib.data
txt: fib.txt
translator_log: fib-1.log
machine_log: fib-2.log
